### PR TITLE
Add valve component

### PR DESCRIPTION
**Pull request by kbx81**
Adds support for valve components/entities

### CODE REVIEW

File: aioesphomeapi/client.py [hunk 1]
```diff
@@ -1154,6 +1155,35 @@ def lock_command(
...
+        key: int,
+        position: float | None = None,
+        stop: bool = False,
+    ) -> None:
+        connection = self._get_connection()
+        req = ValveCommandRequest(key=key)
+        apiv = self.api_version
+        if TYPE_CHECKING:
+            assert apiv is not None
+        if apiv >= APIVersion(1, 1):
```
Comment (bdraco) [inline aioesphomeapi/client.py:1169]: We shouldn't need an api version check since this is new

File: aioesphomeapi/client.py [hunk 2]
```diff
@@ -1154,6 +1155,35 @@ def lock_command(
...
+            if position is not None:
+                req.has_position = True
+                req.position = position
+            if stop:
+                req.stop = stop
+        else:
```
Comment (bdraco) [inline aioesphomeapi/client.py:1175]: Probably can delete this branch

File: aioesphomeapi/client.py (hunk 1 above)
Comment (kbx81) [reply aioesphomeapi/client.py]: Done...

File: aioesphomeapi/client.py (hunk 2 above)
Comment (kbx81) [reply aioesphomeapi/client.py]: ...and done. Thanks @bdraco 🍻

File: aioesphomeapi/api.proto [hunk 3]
```diff
@@ -604,6 +605,74 @@ message TextSensorStateResponse {
...
+  bool assumed_state = 5;
+  bool supports_position = 6;
+  string device_class = 7;
+  bool disabled_by_default = 8;
+  string icon = 9;
+  EntityCategory entity_category = 10;
+  bool supports_stop = 11;
+}
+
+enum LegacyValveState {
```
Comment (bdraco) [inline aioesphomeapi/api.proto:628]: I think we can drop this as well

File: aioesphomeapi/api.proto [hunk 4]
```diff
@@ -604,6 +605,74 @@ message TextSensorStateResponse {
...
+}
+message ValveCommandRequest {
+  option (id) = 107;
+  option (source) = SOURCE_CLIENT;
+  option (ifdef) = "USE_VALVE";
+  option (no_delay) = true;
+
+  fixed32 key = 1;
+
+  // legacy: command has been removed in 1.13
```
Comment (bdraco) [inline aioesphomeapi/api.proto:666]: And this

File: aioesphomeapi/api.proto [hunk 5]
```diff
@@ -604,6 +605,54 @@ message TextSensorStateResponse {
   bool missing_state = 3;
 }
 
+// ==================== VALVE ====================
+message ListEntitiesValveResponse {
+  option (id) = 106;
```
Comment (jesserockz) [inline aioesphomeapi/api.proto:610]:
```suggestion
  option (id) = 109;
```

File: aioesphomeapi/api.proto [hunk 6]
```diff
@@ -604,6 +605,54 @@ message TextSensorStateResponse {
...
+  bool supports_stop = 11;
+}
+
+enum ValveOperation {
+  VALVE_OPERATION_IDLE = 0;
+  VALVE_OPERATION_IS_OPENING = 1;
+  VALVE_OPERATION_IS_CLOSING = 2;
+}
+message ValveStateResponse {
+  option (id) = 108;
```
Comment (jesserockz) [inline aioesphomeapi/api.proto:634]:
```suggestion
  option (id) = 110;
```

File: aioesphomeapi/api.proto [hunk 7]
```diff
@@ -604,6 +605,54 @@ message TextSensorStateResponse {
...
+  option (ifdef) = "USE_VALVE";
+  option (no_delay) = true;
+
+  fixed32 key = 1;
+  float position = 2;
+  ValveOperation current_operation = 3;
+}
+
+message ValveCommandRequest {
+  option (id) = 107;
```
Comment (jesserockz) [inline aioesphomeapi/api.proto:645]:
```suggestion
  option (id) = 111;
```

**Review by jesserockz — CHANGES_REQUESTED** [review]
Please update the below message IDs to not conflict with previous messages and the open Event Entity PR =)

**Review by jesserockz — APPROVED** [review]

**Pull request merged**